test/utils/mocks/MockSampleDelegateCallTarget.sol

```diff
@@ -21,4 +28,14 @@ contract MockSampleDelegateCallTarget {
     function revertWithData(bytes memory data) public pure {
         revert ErrorWithData(data);
     }
+
+    function upgradeHook(bytes32 hookId, string calldata) public returns (bool) {
+        require(hookId == _UPGRADE_HOOK_ID);
```

Comment:
is the `tload` guard basically a cheaper `require msg.sender == address(this)`?

Reply:
Once we’ve made it into a `call`, we can use `msg.sender == address(this)`. 
Cuz `delegatecall` doesn’t change `msg.sender`.

Lemme think if there’s any merit in blocking directly accessing from `execute`.

Reply:
Ok, I think let's keep a transient storage guard for swiss cheese model.